src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -1677,7 +1678,14 @@ def _do_construct_handler(self, node, parent):
         # Loop variable will be an instance of Fortran2003.Name
         loop_var = str(ctrl[0].items[1][0])
         variable_name = str(loop_var)
-        loop = self._create_loop(parent, variable_name)
+        try:
+            data_symbol = parent.find_or_create_symbol(variable_name)
+        except SymbolError:
+            raise InternalError(
+                "Loop-variable name '{0}' is not declared and there are no "
+                "unqualified use statements. This is currently unsupported."
+                "".format(variable_name))
+        loop = self._create_loop(parent, data_symbol)
```

Comment:
This is my fault but could you add the clarifying comment:
```
# The loop node is created with the _create_loop factory method as some
# APIs require a specialised loop node type.
```
before this line.

Reply:
Added.